src/generator/operations.ts

```diff
@@ -575,9 +575,29 @@ function createProtocolRequest(codeModel: CodeModel, op: Operation, imports: Imp
       text += '\treturn req, nil\n';
     }
   } else if (mediaType === 'binary') {
-    const contentType = op.requests![0].protocol.http!.mediaTypes[0];
+    let contentType = `"${op.requests![0].protocol.http!.mediaTypes[0]}"`;
+    if (op.requests![0].protocol.http!.mediaTypes.length > 1) {
+      for (const param of values(op.requests![0].parameters)) {
+        // If a request defined more than one possible media type, then the param is expected to be synthesized from modelerfour
+        // and should be a SealedChoice schema type that account for the acceptable media types defined in the swagger. 
+        if (param.origin !== undefined && param.origin === 'modelerfour:synthesized/content-type' && param.schema.type === SchemaType.SealedChoice) {
+          contentType = `string(${param.language.go!.name})`;
+        }
+      }
+    }
     const bodyParam = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http!.in === 'body'; }).first();
-    text += `\treturn req, req.SetBody(${bodyParam?.language.go!.name}, "${contentType}")\n`;
+    text += `\treturn req, req.SetBody(${bodyParam?.language.go!.name}, ${contentType})\n`;
+  } else if (mediaType === 'text') {
+    imports.add('strings');
+    let bodyParam = '';
+    for (const param of values(op.requests![0].parameters)) {
+      if (param.protocol.http!.in !== undefined && param.protocol.http!.in === 'body') {
+        bodyParam = param.language.go!.name;
+      }
+    }
+    text += `\treq.Header.Set("Content-Type", "text/plain; encoding=UTF-8")\n`;
```

Comment:
You can remove this line, the `content-type` is set in the call to `req.SetBody()`.

Reply:
I had it without explicitly setting the header originally but it seems there could be a problem in SetBody because the test fails when reading the content-type header without it being explicitly set like this

Reply:
I'll test it out once more without it to see, there's probably something that needs to be fixed in SetBody, I havent looked at it yet

Reply:
We should fix `SetBody()` then (the whole point of that change was to set the content-type there).

Reply:
It's because you're sending a string of zero length, so the header (and body) is omitted.  Does the test expect a zero-length string?

Reply:
https://github.com/Azure/autorest.testserver/blob/faa78f54f3e42d51dd001cb2644c2139244114a1/legacy/routes/mediatypes.js#L32

I see. The test doesnt care, I'll send some random string then for the value.

Reply:
You can send any length string.

Reply:
Done